src/app/clusters/camera-av-stream-management-server/camera-av-stream-management-server.h

```diff
@@ -0,0 +1,636 @@
+#include <vector>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace CameraAvStreamManagement {
+
+using VideoStreamStruct            = Structs::VideoStreamStruct::Type;
+using AudioStreamStruct            = Structs::AudioStreamStruct::Type;
+using SnapshotStreamStruct         = Structs::SnapshotStreamStruct::Type;
+using AudioCapabilitiesStruct      = Structs::AudioCapabilitiesStruct::Type;
+using VideoSensorParamsStruct      = Structs::VideoSensorParamsStruct::Type;
+using SnapshotParamsStruct         = Structs::SnapshotParamsStruct::Type;
+using VideoResolutionStruct        = Structs::VideoResolutionStruct::Type;
+using ViewportStruct               = Structs::ViewportStruct::Type;
+using RateDistortionTradeOffStruct = Structs::RateDistortionTradeOffPointsStruct::Type;
+using SnapshotParamsStruct         = Structs::SnapshotParamsStruct::Type;
+
+constexpr uint8_t kMaxSpeakerLevel          = 254;
+constexpr uint8_t kMaxMicrophoneLevel       = 254;
+constexpr uint16_t kMaxImageRotationDegrees = 359;
+constexpr uint8_t kMaxChannelCount          = 8;
+constexpr uint8_t kMaxImageQualityMetric    = 100;
+constexpr uint16_t kMaxFragmentLenMaxValue  = 65500;
+// Conservative room for other fields (resolution + codec) in
+// capture snapshot response. TODO: Make a tighter bound.
+constexpr size_t kMaxSnapshotImageSize = kMaxLargeSecureSduLengthBytes - 100;
+
+constexpr size_t kViewportStructMaxSerializedSize =
+    TLV::EstimateStructOverhead(sizeof(uint16_t), sizeof(uint16_t), sizeof(uint16_t), sizeof(uint16_t));
+
+// The number of possible values of StreamUsageEnum.
+constexpr size_t kNumOfStreamUsageTypes = 4;
+
+// StreamUsageEnum + Anonymous tag (1 byte).
+// Assumes min-size encoding (1 byte) for the integer.
+constexpr size_t kStreamUsageTlvSize = sizeof(StreamUsageEnum) + 1;
+
+// 1 control byte + end-of-array marker
+constexpr size_t kArrayTlvOverhead = 2;
+
+constexpr size_t kRankedVideoStreamPrioritiesTlvSize = kArrayTlvOverhead + kStreamUsageTlvSize * kNumOfStreamUsageTypes;
+
+class CameraAVStreamMgmtServer;
+
+// ImageSnapshot response data for a CaptureSnapshot command.
+struct ImageSnapshot
+{
+    std::vector<uint8_t> data;      // Buffer to hold the image data
+    VideoResolutionStruct imageRes; // Image resolution
+    ImageCodecEnum imageCodec;      // Image codec used
+};
+
+/** @brief
+ *  Defines interfaces for implementing application-specific logic for various aspects of the CameraAvStreamManagement Cluster.
+ *  Specifically, it defines interfaces for the command handling and loading of the allocated streams.
+ */
+class CameraAVStreamMgmtDelegate
+{
+public:
+    CameraAVStreamMgmtDelegate() = default;
+
+    virtual ~CameraAVStreamMgmtDelegate() = default;
+
+    /**
+     *   @brief Handle Command Delegate for Video stream allocation with the provided parameter list.
+     *
+     *   @param[in]  allocateArgs   Structure with parameters for video stream allocation.
```

Comment:
Should document that callee must ignore the videoStreamID and referenceCount values?  Similar for the other Allocate methods.